src/bridge/types.js

```diff
@@ -76,15 +76,16 @@ export interface WalletBridge<Transaction> {
 
   getTransactionRecipient(account: Account, transaction: Transaction): string;
 
-  isValidTransaction(account: Account, transaction: Transaction): boolean;
-
   // render the whole Fees section of the form
   EditFees?: *; // React$ComponentType<EditProps<Transaction>>;
 
   // render the whole advanced part of the form
   EditAdvancedOptions?: *; // React$ComponentType<EditProps<Transaction>>;
 
-  checkCanBeSpent(account: Account, transaction: Transaction): Promise<void>;
+  // validate the transaction and all currency specific validations here, we can return false
+  // to disable the button without throwing an error if we are handling the error on a different
+  // input or throw an error that will highlight the issue on the amount field
```

Comment:
Please detail why we need the case of returning "false"? I think we always want to give feedback to user with an error (in failing promise) to inform why it's blocking.
and if we don't, we probably just should ignore a given error to not display it.

NB: i'm currently in progress of refactoring all of these functions in our mobile implementation in order to unify things a bit better between the 2 projects for the future, so we might want to hodl a bit on this.

Reply:
If I throw an exception on the `checkValidTransaction` method, the field that gets the feedback and message is the amount field. I wanted to show the error on a different field but still disable the _next_ button if the validation failed.

  ```
    checkValidTransaction: async (a, t) => {
      if (t.memoError) {
        return false
      }
      if (t.amount.isLessThanOrEqualTo(a.balance.plus(t.fee))) {
        return true
      }
      throw new NotEnoughBalance()
    }
```
I'm not against changing this at all, I just needed a way to throw an error on my own input from the `AdvancedOptions` without it being linked to the amount field. So with the code in this pr, the validation of an invalid form, failing because of a memo issue, would give us this

![image](https://user-images.githubusercontent.com/4631227/44909964-afd07380-ad21-11e8-85f4-369100897528.png)


If instead of relying on the false case like I'm doing I throw an error if `t.memoError` I get this sort of validation which is not really true because I have no issue with the value of amount but rather the value of the memo. Might be a better way that I'm failing to see?

![image](https://user-images.githubusercontent.com/4631227/44910116-47ce5d00-ad22-11e8-8716-b7e974bfb4ad.png)